GitExtUtils/FileLogging/LogFilenameCreator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.IO;
+
+namespace GitExtUtils.FileLogging
+{
+    public class LogFilenameCreator
+    {
+        private readonly string _logName;
+        public readonly string LogPath;
+
+        private Tuple<DateTime, string> _result;
+
+        public LogFilenameCreator(string logPath, string logName)
+        {
+            LogPath = logPath;
+            _logName = logName;
+            _result = Renew();
+        }
+
+        private Tuple<DateTime, string> Renew()
+        {
+            var today = DateTime.Today;
+            var fullPath = Path.Combine(LogPath, $"{_logName}-{today:yyyy-MM-dd}.log");
+            return Tuple.Create(today, fullPath);
+        }
+
+        public string Create()
+        {
+            if (DateTime.Today != _result.Item1)
+            {
+                _result = Renew();
+            }
+
+            return _result.Item2;
+        }
+
+        public string GetCleaningFilenamePattern()
```

Comment:
This is a `LogFilenameCreator`, it has nothing to do with "cleaning".
```suggestion
        public string GetFilenamePattern()
```

Also why does it have to be a method?

Reply:
Good name suggestion. 

It is a method because it performs computations. I detest properties that have side effects, be it state changes or "CPU consumption".